hmda-regulator/src/main/scala/hmda/regulator/scheduler/LarScheduler.scala

```diff
@@ -80,24 +82,29 @@ class LarScheduler extends HmdaActor with RegulatorComponent {
       val s3Sink =
         s3Client.multipartUpload(bucket, s"$environment/lar/$fileName")
 
-      val allResults: Future[Seq[LarEntityImpl]] =
-        larRepository.getAllLARs(bankFilterList)
-
-      val results: Future[MultipartUploadResult] = Source
-        .fromFuture(allResults)
-        .map(seek => seek.toList)
-        .mapConcat(identity)
-        .map(larEntity => larEntity.toPSV + "\n")
-        .map(s => ByteString(s))
-        .runWith(s3Sink)
-
-      results onComplete {
-        case Success(result) => {
-          log.info(
-            "Pushing to S3: " + s"$bucket/$environment/lar/$fileName" + ".")
+      try {
+        val allResultsPublisher: DatabasePublisher[LarEntityImpl] =
+          larRepository.getAllLARs(bankFilterList)
+        val allResultsSource: Source[LarEntityImpl, NotUsed] =
+          Source.fromPublisher(allResultsPublisher)
+
+        var results: Future[MultipartUploadResult] = allResultsSource
+          .map(larEntity => larEntity.toPSV + "\n")
+          .map(s => ByteString(s))
+          .runWith(s3Sink)
+
+        results onComplete {
+          case Success(result) => {
+            log.info(
+              "Pushing to S3: " + s"$bucket/$environment/lar/$fileName" + ".")
+          }
+          case Failure(t) =>
+            println("An error has occurred getting LAR Data in Future: " + t.getMessage)
+
         }
-        case Failure(t) =>
-          println("An error has occurred getting LAR Data: " + t.getMessage)
+      } catch {
```

Comment:
This is minor but generally the more scala-y approach to error handling is to use `try` and then `case Success` and `case Failure` instead of `catch`. Here is a [short article](https://www.codacy.com/blog/why-use-try-instead-of-try-catch-in-scala/) on it. I think it might be a good change just to keep our codebase consistent.

Reply:
This try and catch block is a relic from testing/implementing the streaming logic. It's actually redundant now that everything is in place and we're tracking the Futures completion via case Success/Failure